client/scripts/models/SearchQuery.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable prefer-template */
+export enum SearchScope {
+    "Members" = "Members",
+    "Communities" = "Communities",
+    "Proposals" = "Proposals",
+    "Threads" = "Threads",
+    "Replies" = "Replies",
+    "All" = "All"
+}
+
+export enum SearchSort {
+    "Best" = "Best",
+    "Newest" = "Newest",
+    "Oldest" = "Oldest"
+}
+export interface SearchParams {
+    communityScope?: string;
+    chainScope?: string;
+    isSearchPreview?: boolean;
+    searchScope?: Array<SearchScope>;
+    sort?: SearchSort;
+    resultSize?: number;
+}
+export default class SearchQuery implements SearchParams {
+    public searchTerm: Lowercase<string>;
+    public communityScope?: string;
+    public chainScope?: string;
+    public isSearchPreview?: boolean;
+    public searchScope: Array<SearchScope>;
+    public sort: SearchSort;
+
+    constructor(searchTerm: string, params?: SearchParams){
+        this.searchTerm = searchTerm
+        this.searchScope = params?.searchScope || [SearchScope.All]
+        this.communityScope = params?.communityScope
+        this.chainScope = params?.chainScope
+        this.isSearchPreview = !!params?.isSearchPreview
+        this.sort = params?.sort || SearchSort.Best
+    }
+
+    public toEncodedString() {
+        let encodedString = this.searchTerm.trim().replace(/\s+/g, '%20') +
+            (this.communityScope ? ` communityScope=${this.communityScope}` : '') +
+            (this.chainScope ? ` chainScope=${this.chainScope}` : '') +
+            (this.isSearchPreview ? ` isSearchPreview=${this.isSearchPreview}` : '') +
+            (this.sort ? ` sort=${this.sort}` : '')
+
+        for(const scope in this.searchScope){
+            if (Object.prototype.hasOwnProperty.call(this.searchScope, scope) &&
+                this.searchScope[scope] !== SearchScope.All) {
+                encodedString += ` scope[]=${this.searchScope[scope]}`
+            }
+        }
+        return encodedString
+    }
+
+    public static fromEncodedString(encodedString: string) {
+        const props = encodedString.split(" ")
+        const sq = new SearchQuery(props[0].replace(/(%20)/g, ' '))
+        for(let i = 1; i < props.length; i++){
```

Comment:
Because props is an iterable, this for loop can be simplified to: 
`for (const prop in props) { ... } `

Reply:
This loop has specific behavior for the first item in the list, so it would be possible to switch to an advanced loop, but would require extra work (slicing or unshifting) - probably best as is?